SpatialGDK/Source/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -466,131 +510,86 @@ void USpatialActorChannel::SetChannelActor(AActor* InActor)
 	}
 
 	// Set up the shadow data for the handover properties. This is used later to compare the properties and send only changed ones.
-	USpatialInterop* Interop = SpatialNetDriver->GetSpatialInterop();
-	check(Interop);
-	USpatialTypeBinding* Binding = Interop->GetTypeBindingByClass(InActor->GetClass());
-	if (Binding)
-	{
-		const FHandoverHandlePropertyMap& HandoverProperties = Binding->GetHandoverHandlePropertyMap();
-		uint32 Size = 0;
-		for (auto& Property : HandoverProperties)
-		{
-			// Make sure we conform to Unreal's alignment requirements; this is matched below and in ReplicateActor()
-			Size = Align(Size, Property.Value.Property->GetMinAlignment());
-			Size += Property.Value.Property->GetSize();
-		}
-		HandoverPropertyShadowData.Empty();
-		HandoverPropertyShadowData.AddZeroed(Size);
-		uint32 Offset = 0;
-		for (auto& Property : HandoverProperties)
-		{
-			Offset = Align(Offset, Property.Value.Property->GetMinAlignment());
-			Property.Value.Property->InitializeValue(HandoverPropertyShadowData.GetData() + Offset);
-			Offset += Property.Value.Property->GetSize();
-		}
+	check(!HandoverShadowDataMap.Contains(InActor));
+
+	// Create the shadow map, and store a quick access pointer to it
+	ActorHandoverShadowData = &HandoverShadowDataMap.Add(InActor, MakeShared<TArray<uint8>>()).Get();
+	InitializeHandoverShadowData(*ActorHandoverShadowData, InActor);
+
+	// Assume that all the replicated static components are already set as such. This is checked later in ReplicateSubobject.
+	for (UObject* Subobject : NetDriver->TypebindingManager->GetHandoverSubobjects(InActor))
+	{
+		check(!HandoverShadowDataMap.Contains(Subobject));
+		InitializeHandoverShadowData(HandoverShadowDataMap.Add(Subobject, MakeShared<TArray<uint8>>()).Get(), Subobject);
 	}
 
 	// Get the entity ID from the entity registry (or return 0 if it doesn't exist).
-	check(SpatialNetDriver->GetEntityRegistry());
-	ActorEntityId = SpatialNetDriver->GetEntityRegistry()->GetEntityIdFromActor(InActor).ToSpatialEntityId();
+	check(NetDriver->GetEntityRegistry());
+	EntityId = NetDriver->GetEntityRegistry()->GetEntityIdFromActor(InActor);
 
 	// If the entity registry has no entry for this actor, this means we need to create it.
-	if (ActorEntityId == 0)
+	if (EntityId == 0)
```

Comment:
Is there a constant for this in the worker api? If not we should define this in Spatial constants.

Reply:
Oh - SpatialConstants::INVALID_ENTITY_ID